.github/workflows/deploy.yaml

```diff
@@ -29,45 +27,131 @@ jobs:
+
+      - name: Check if cypress test exists
+        id: find-cypress
         run: |
-          cat > /usr/local/lib/R/etc/Rprofile.site <<RPROFILE
-           options(
-            repos = c(
-              Pharmaverse = "https://pharmaverse.r-universe.dev",
-              CRAN = "https://cloud.r-project.org",
-              BioC = BiocManager::repositories()
-            ),
-            download.file.method = 'libcurl',
-            HTTPUserAgent = sprintf(
-              "R/%s R (%s)",
-              getRversion(),
-              paste(getRversion(), R.version["platform"], R.version["arch"], R.version["os"])
-            )
-          )
-          RPROFILE
+          if [ -d ${{ github.event.repository.name }}/${{ matrix.directory }}/tests/cypress ]; then
+            echo "has-cypress-tests=true" >> $GITHUB_OUTPUT
+          else
+            echo "Skipping frontend tests because cypress directory does not exist."
+            echo "has-cypress-tests=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Setup system dependencies for cypress and python app
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        run: >
+          apt-get update && apt-get install --yes
+          libgtk2.0-0 libgbm-dev libnotify-dev libgconf-2-4 xvfb python3.10-venv
 
-      - name: Deploy apps 🖨 ${{ github.event.repository.name }} 🎨
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        with:
+          node-version: 16
+
+      - name: Setup npm packages
+        if: steps.find-cypress.outputs.has-cypress-tests == 'true'
+        shell: bash
+        working-directory: ${{ github.event.repository.name }}/${{ matrix.directory }}/js
         run: |
-          Sys.setenv(GITHUB_PAT="${{ secrets.GITHUB_TOKEN }}")
-          devtools::install_local(
-            ".",
-            upgrade = "never",
-            dependencies = TRUE
-          )
-          teal.gallery::deploy_all_apps(
-            api_key = "${{ secrets.SHINYAPPSIO_TOKEN }}",
-            api_secret = "${{ secrets.SHINYAPPSIO_SECRET }}",
-            title_prefix = "NEST",
-            title_suffix = "${{ github.ref_name }}",
-            server = "https://${{ env.SHINYAPPSIO_ACCOUNT }}.shinyapps.io",
-            create_manifest = TRUE
-          )
+          npm i
+
+      - name: Restore renv from cache
+        uses: actions/cache@v2
+        env:
+          CACHE_KEY: renv-${{ runner.arch }}-${{ runner.os }}-${{ env.R_VERSION }}-${{ matrix.directory }}-${{ matrix.channel }}
+        with:
+          path: ${{ github.event.repository.name }}/${{ matrix.directory }}/renv/library
+          key: ${{ env.CACHE_KEY }}-${{ hashFiles(format('{0}/{1}/renv.lock', github.event.repository.name, matrix.directory)) }}
+          restore-keys: ${{ env.CACHE_KEY }}-
```

Comment:
@pawelru Do you see some problem with the restore cache step?

For some reason, the restored cache does not seem to help `renv::restore()` take advantage of the cache. I remember it used to work earlier not anymore.

Reply:
I haven't looked into the caching so far... Maybe worth to ask Dinakar or Franek for some help on that?